Source/UI/NewUI/BillingForm/Factory/BillingFormFactory.swift

```diff
@@ -0,0 +1,22 @@
+import UIKit
+
+struct BillingFormFactory {
+    // in order
+    static let styles: [BillingFormTextFieldCellStyle] = [
```

Comment:
Could these styles just be an enum case?

Lets imagine:
```
enum BillingFormTextFieldCellStyle {
    case fullName
    case addressLine1
    .....

    func getConfiguration() -> BilingFormTextFieldConfig {
        switch self {
        case .fullName:
            return .init(color1: .... , color2: ...)
        }
    }
}
```

My initial thought is it would reduce a lot of the syntactic sugar you currently have and may increase configuration options as well as extensibility. Additionally, you could completely drop this `styles` property if you make the enum declaration conform to `CaseIterable` and end up with a syntax of `.allCases` at call site instead of the current `styles` reference

Reply:
@ehab-al-cko to Create Tech investment ticket to address this in new PR